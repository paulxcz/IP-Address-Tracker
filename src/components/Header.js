import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
import { IpInfo } from "./IpInfo";


import "../styles/Header/Header.css";
export default function Header({setcords}) {

  const [ipToFind, setipToFind] = useState({
    visibility : false,
    ip : "",
  });

  const { visibility, ip } = ipToFind;

  
  return (
    <>
      <header>
        <h1>IP Address Tracker</h1>
        <div className="search-box">
          <input type="text"
            value = {ip}
            onChange = { ({target}) => { setipToFind({
              visibility: false,
              ip: target.value
            }) }}
            maxLength="15"
            placeholder = "Search for any IP adress or domain"
            />
          <button
            onClick={(e) => {
              e.preventDefault();
              setipToFind({
                ...ipToFind,
                visibility: !visibility,
              });
            }}
          >
            <FontAwesomeIcon icon={faChevronRight} className="right-arrow" />
          </button>
        </div>

        {visibility && <IpInfo ipToFind={ip} setcords={setcords}/>}
      </header>
    </>
  );
}
