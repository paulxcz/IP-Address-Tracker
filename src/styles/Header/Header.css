input[type="number"]::-webkit-inner-spin-button,
input[type="number"]::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

input[type="number"] {
  -moz-appearance: textfield;
}

header {
  background: url("../../images/pattern-bg.png");
  min-height: 256px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

header h1 {
  font-size: 22px;
  color: #fff;
  margin-top: 27px;
}

header .search-box {
  margin-top: 30px;
  min-width: 250px;
  width: 85%;
  max-width: 554px;
  height: 50px;
  background: #000;
  border-radius: 10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
}

header .search-box input {
  width: 80%;
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
  border: none;
  font-size: 17px;
  padding: 0 20px;
  cursor: pointer;
}

header .search-box button {
  border: none;
  background-color: #000;
  outline: none;
  display: block;
  width: 20%;
  cursor: pointer;
}

header .search-box button:hover {
  background-color: #3F3F3F;
}

header .search-box button .right-arrow {
  color: #fff;
}
/*# sourceMappingURL=Header.css.map */