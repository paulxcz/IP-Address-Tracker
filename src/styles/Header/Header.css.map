{
    "version": 3,
    "mappings": "AAAA,AAAA,KAAK,CAAA,AAAA,IAAC,CAAK,QAAQ,AAAb,CAAc,2BAA2B;AAC/C,KAAK,CAAA,AAAA,IAAC,CAAK,QAAQ,AAAb,CAAc,2BAA2B,CAAC;EAC9C,kBAAkB,EAAE,IAAI;EACxB,MAAM,EAAE,CAAC;CACV;;AACD,AAAA,KAAK,CAAA,AAAA,IAAC,CAAK,QAAQ,AAAb,EAAe;EACnB,eAAe,EAAE,SAAS;CAC3B;;AAGD,AAAA,MAAM,CAAC;EACL,UAAU,EAAE,kCAAkC;EAC9C,UAAU,EAAE,KAAK;EACjB,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,WAAW,EAAE,MAAM;CA2CpB;;AAhDD,AAME,MANI,CAMJ,EAAE,CAAC;EACD,SAAS,EAAE,IAAI;EACf,KAAK,EAAE,IAAI;EACX,UAAU,EAAE,IAAI;CACjB;;AAVH,AAWE,MAXI,CAWJ,WAAW,CAAC;EACV,UAAU,EAAE,IAAI;EAChB,SAAS,EAAE,KAAK;EAChB,KAAK,EAAE,GAAG;EACV,SAAS,EAAE,KAAK;EAChB,MAAM,EAAE,IAAI;EACZ,UAAU,EAAE,IAAI;EAChB,aAAa,EAAE,IAAI;EACnB,OAAO,EAAE,IAAI;EACb,QAAQ,EAAE,MAAM;CA2BjB;;AA/CH,AAuBI,MAvBE,CAWJ,WAAW,CAYT,KAAK,CAAC;EACJ,KAAK,EAAE,GAAG;EACV,sBAAsB,EAAE,IAAI;EAC5B,yBAAyB,EAAE,IAAI;EAC/B,MAAM,EAAE,IAAI;EACZ,SAAS,EAAE,IAAI;EACf,OAAO,EAAE,MAAM;EACf,MAAM,EAAE,OAAO;CAChB;;AA/BL,AAgCI,MAhCE,CAWJ,WAAW,CAqBT,MAAM,CAAC;EACL,MAAM,EAAE,IAAI;EACZ,gBAAgB,EAAE,IAAI;EACtB,OAAO,EAAE,IAAI;EACb,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,GAAG;EACV,MAAM,EAAE,OAAO;CAQhB;;AA9CL,AAwCM,MAxCA,CAWJ,WAAW,CAqBT,MAAM,AAQH,MAAM,CAAA;EACL,gBAAgB,EAAE,OAAO;CAC1B;;AA1CP,AA2CM,MA3CA,CAWJ,WAAW,CAqBT,MAAM,CAWJ,YAAY,CAAC;EACX,KAAK,EAAE,IAAI;CACZ",
    "sources": [
        "Header.scss"
    ],
    "names": [],
    "file": "Header.css"
}