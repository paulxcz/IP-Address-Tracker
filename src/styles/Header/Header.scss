input[type="number"]::-webkit-inner-spin-button,
input[type="number"]::-webkit-outer-spin-button {
  -webkit-appearance: none;
  margin: 0;
}
input[type="number"] {
  -moz-appearance: textfield;
}


header {
  background: url("../../images/pattern-bg.png");
  min-height: 256px;
  display: flex;
  flex-direction: column;
  align-items: center;
  h1 {
    font-size: 22px;
    color: #fff;
    margin-top: 27px;
  }
  .search-box {
    margin-top: 30px;
    min-width: 250px;
    width: 85%;
    max-width: 554px;
    height: 50px;
    background: #000;
    border-radius: 10px;
    display: flex;
    overflow: hidden;
    // align-items: center;
    // justify-content: space-between;
    input {
      width: 80%;
      border-top-left-radius: 10px;
      border-bottom-left-radius: 10px;
      border: none;
      font-size: 17px;
      padding: 0 20px;
      cursor: pointer;
    }
    button {
      border: none;
      background-color: #000;
      outline: none;
      display: block;
      width: 20%;
      cursor: pointer;

      &:hover{
        background-color: #3F3F3F;
      }
      .right-arrow {
        color: #fff;
      }
    }
  }
}
