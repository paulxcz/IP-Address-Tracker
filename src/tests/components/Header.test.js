import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import Header from '../../components/Header';

describe('Tests on <Header />', () => {

    const setCords = jest.fn();
    let wrapper = shallow(<Header setcords={setCords}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<Header setcords={setCords}/>);
    });
    test('should display <Header /> correct', () => {
        
        expect( wrapper ).toMatchSnapshot( );
    })
        
    test('should not to show <IpInfo /> when input "change" ', () => {

        wrapper.find('input').simulate('change', {target: 'TextOnInput'});
        expect( wrapper.find('IpInfo').exists() ).toBe(false);
    })

    test('should show <IpInfo /> when press <button /> to show IP info ', () => {

        wrapper.find('button').simulate('click', {preventDefault: () =>{} });
        expect( wrapper.find('IpInfo').exists() ).toBe(true);
    })
})