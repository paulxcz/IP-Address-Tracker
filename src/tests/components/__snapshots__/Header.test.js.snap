// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Header /> should display <Header /> correct 1`] = `
<Fragment>
  <header>
    <h1>
      IP Address Tracker
    </h1>
    <div
      className="search-box"
    >
      <input
        maxLength="15"
        onChange={[Function]}
        placeholder="Search for any IP adress or domain"
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
      >
        <FontAwesomeIcon
          border={false}
          className="right-arrow"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                320,
                512,
                Array [],
                "f054",
                "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
              ],
              "iconName": "chevron-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
  </header>
</Fragment>
`;
